DOCS-6706 Metaspace constraints (SE-14891) (#291)
mulesoft/docs-runtime-manager: modules/ROOT/pages/cloudhub-architecture.adoc

endif::[]
:keywords: cloudhub, cloud, manage, mmc, api, worker, runtime manager, arm

image:logo-cloud-active.png[link="/runtime-manager/deployment-strategies", title="CloudHub"]
image:logo-hybrid-disabled.png[link="/runtime-manager/deployment-strategies", title="Hybrid"]
image:logo-server-disabled.png[link="/runtime-manager/deployment-strategies", title="Anypoint Platform PCE"]
image:logo-rtf-disabled.png[link="/runtime-manager/deployment-strategies", title="Runtime Fabric"]

> xref:cloudhub.adoc[CloudHub] has been built from the ground up to provide enterprises with a multi-tenant, secure, elastic, and highly available integration platform as a service (iPaaS). This document describes how the underlying mechanisms of the CloudHub platform work to achieve these goals.

Reviewer: Well, since you mentioned the white...:
```suggestion
xref:cloudhub.adoc[CloudHub] is designed to provide enterprises with a multitenant, secure, elastic, and highly available integration platform as a service (iPaaS). To maximize your use of CloudHub, It is important to understand how the underlying mechanisms of the CloudHub platform work to achieve these goals.
```

> Manage CloudHub with the Runtime Manager console in the Anypoint Platform. You can also xref:deploying-to-cloudhub.adoc[deploy] to it directly from Anypoint Studio, via the xref:cloudhub-api.adoc[CloudHub API] or via the xref:anypoint-platform-cli.adoc[Anypoint Platform Command Line Interface (Anypoint CLI)].

Reviewer: ```suggestion
Manage CloudHub with the Anypoint Runtime Manager console in Anypoint Platform. You can also xref:deploying-to-cloudhub.adoc[deploy] to it directly from Anypoint Studio, via the xref:cloudhub-api.adoc[CloudHub API] or via the xref:anypoint-platform-cli.adoc[Anypoint Platform command-line interface].
```

=== CloudHub Workers

Applications on CloudHub are run by one or more instances of Mule, called *workers*.
Each worker is a dedicated instance of Mule that runs your integration application.

Workers have the following characteristics:

*Capacity*::
Each worker has a specific amount of capacity to process data. Select the size of your workers when configuring an application.
*Isolation*::
Each worker runs in a separate container from every other application.
*Manageability*::
Each worker is deployed and monitored independently.
*Locality*::
Each worker runs in a specific worker cloud, such as the US, EU, or Asia-Pacific.

> The memory capacity and processing power of a worker depends on how you configure them at the application level. 

Reviewer: ```suggestion
The memory capacity and processing power of a worker depends on how you configure it at the application level. 
```

* 8 vCores + 15 GB Heap Memory
* 16 vCores + 32 GB Heap Memory

> For workers that use a fraction of a vCore, CloudHub allocates limited CPU and IO as well as less memory. 

Reviewer: ```suggestion
For workers that use a fraction of a vCore, CloudHub allocates limited CPU and I/O, as well as less memory. 
```

Use these worker sizes for apps with smaller workloads.

> For apps deployed to CloudHub with worker sizes of 0.1, 0.2, and 1 vCore, the metaspace limit is 256 MB.

Reviewer 2: For the current documentation, we should just mention that for apps deployed to Cloudhub, the metaspace limit is 256MB.

> MuleSoft might change this limit in the future.

Reviewer: "The metaspace limit for apps deployed to CloudHub is currently 256 MB, regardless of the worker size."

We should never mention something that might happen in the future.

|Asia Pacific (Singapore)| `sg-s1.cloudhub.io`
|===

The domain provided for your application is based on the region to which your application is deployed. For example, if you deploy an application named `myapp` to Canada (Central), the domain used to access the application is `myapp.ca-c1.cloudhub.io`. The load balancer used to route requests resides in the same region as your application.

> For more information about deploying applications to different regions, refer to xref:deploying-to-cloudhub.adoc[Deploying to CloudHub]. For more information about CloudHub's security and compliance, download the https://www.mulesoft.com/lp/whitepaper/saas/cloud-security[Anypoint Cloud Security & Compliance whitepaper].

Reviewer: Does the white paper still exist?

Author: Yes.

> CloudHub monitors the worker clouds for any type of problems and provides a self-healing mechanism to recover from problems. If the underlying hardware suffers a failure, the platform migrates your application to a new worker automatically. In the case of an application crash – whether due to a problem with custom code or a bug in the underlying stack – the platform recognizes the crash and can xref:worker-monitoring.adoc#enabling-automatic-restarts[restart the worker automatically].

Reviewer: ```suggestion
CloudHub monitors the worker clouds for problems and provides a self-healing mechanism to recover from them. If the underlying hardware suffers a failure, the platform migrates your application to a new worker automatically. In the case of an application crash, whether due to a problem with custom code or a bug in the underlying stack, the platform recognizes the crash and can xref:worker-monitoring.adoc#enabling-automatic-restarts[restart the worker automatically].
```
